sdk/program/src/pubkey.rs

```diff
@@ -740,6 +740,25 @@ macro_rules! impl_borsh_serialize {
 impl_borsh_serialize!(borsh0_10);
 impl_borsh_serialize!(borsh0_9);
 
+/*
+ * #[cfg(RUSTC_WITH_SPECIALIZATION)]
+ * impl solana_frozen_abi::abi_example::AbiExample for Pubkey {
+ *     fn example() -> Self {
+ *         log::info!("AbiExample for &Pubkey: {}", std::any::type_name::<Self>());
+ *         const EXAMPLE: Pubkey = Pubkey([0; 32]);
+ *         EXAMPLE
+ *     }
+ * }
+ */
+
+#[cfg(RUSTC_WITH_SPECIALIZATION)]
+impl solana_frozen_abi::abi_example::AbiExample for &Pubkey {
+    fn example() -> Self {
+        log::info!("AbiExample for &Pubkey: {}", std::any::type_name::<Self>());
+        &Pubkey([0; 32])
+    }
+}
```

Comment:
> @ryoqun Do you happen to know a better/more-correct way to handle this? I'd prefer not to change the type on the `owner` field to be `&[u8; 32]`.

@brooksprumo yeah. hope this commit is self-explanatory. :) happy to explain a bit if desired: https://github.com/solana-labs/solana/pull/35118/commits/c0eaaf5f732f34836ae8c6a4d67782cd93e7ad80

> which does seem to get called, but I'm not sure how to connect the dots with `&Pubkey`...

I understand your feeling. These referenced visitor code is absurdly complex due to various restrictions, even to today's me.... ;) As a context, such huge effort has been put so that 99% of time you just need proper `AbiExample` trait impls. After several years, i still think frozen_abi is one of darkest magics, which I've ever wrote.

Sorry for being late to reply... Please chime in me for any trouble of frozen abi..